contracts/sg721/src/state.rs

```diff
@@ -1,15 +1,14 @@
-use crate::ContractError;
 use cosmwasm_std::{Addr, Decimal};
 use cw_storage_plus::Item;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-pub struct CollectionInfo {
+pub struct CollectionInfo<T> {
     pub description: String,
     pub image: String,
     pub external_link: Option<String>,
-    pub royalties: Option<RoyaltyInfo>,
+    pub royalty_info: Option<T>,
```

Comment:
This is _really_ nice. Will provide a lot more flexibility in the future. :heart: